### ISSUE TITLE
Avoid DoS potential on dashboard due to large ci_logs table

### ISSUE THREAD

**Issue by adborden** [GSA/data.gov#535]
We're seeing "Service Unavailable" on the crm web service as well as background jobs failing without much information. In looking closer at the database, mysql client hangs when connecting as it gathers statistics about the database. If you look at the tables individually `ci_logs` seems to be causing this hang. This is an [Archive](https://dev.mysql.com/doc/refman/5.7/en/archive-storage-engine.html) table, so there is no index on it.

Related https://github.com/GSA/datagov-deploy/issues/533

**Comment by adborden** [comment]
@philipashlock From what I can tell, this table is just used to store profiling information, it logs every request and is never cleaned up.

Are we reading this information for anything? I think for staging, we can just truncate the table and that will resolve the issue. For production, we may want to truncate the table, but also set the log frequency to 1 percent based on traffic.

**Comment by adborden** [comment]
This just hit us on production Dashboard.

### Timeline
- 9/3 noticed no data on Dashboard pages
- 9/11 confirmed that this is an issue
- confirmed high number of connections to database "Too many connections" error
- BSP restarted mysql database
- BSP identified high connection count
![image](https://user-images.githubusercontent.com/509703/64733814-476eb300-d49a-11e9-8770-2d4f6c25ce57.png)
- confirmed high number of connections to PHP and mysql `lsof -i | egrep '9000|mysql'`
- connection to database via `mysql` was slow, reminding us of the ci_logs issue
- `select count(*) from ci_logs` hung for at least a minute before we killed it
- `truncate table ci_logs` failed, with error `ERROR 1034 (HY000): Incorrect key file for table 'ci_logs'; try to repair it`. Archive tables also do not support TRUNCATE. Instead, DROP the table and the Dashboard app will [re-create it](https://github.com/GSA/project-open-data-dashboard/blob/baa220f6f069fe627a2b7180bc0f28cf2f0a2b53/application/libraries/CILogger.php#L144).
- Issued `drop table ci_logs`
- Requests to dashboard seem much faster

**Comment by mogul** [comment]
For when/if we address this: I developed a little insight on this problem via [related archaeology](https://github.com/GSA/project-open-data-dashboard/pull/218#discussion_r352996958).

**Comment by mogul** [comment]
[Further discussion](https://github.com/GSA/project-open-data-dashboard/pull/218#discussion_r356740996) on how to address this.

TL;DR: We need to determine if there's a need for this granularity of logging.

- If not, we should rip CILogger out.
- If so, we should modify CILogger to write to normal logging output streams, and rely on log rotation to handle this issue.

**Comment by jbrown-xentity** [comment]
Related to dashboard application, which has been deprecated. Closing.
